org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -195,7 +195,7 @@ public static String getPackageName(IJavaProject javaProject, URI uri) {
 			if (content.isEmpty() && javaProject != null && ProjectsManager.DEFAULT_PROJECT_NAME.equals(javaProject.getProject().getName())) {
 				java.nio.file.Path path = Paths.get(uri);
 				java.nio.file.Path parent = path;
-				while (parent.getParent() != null) {
+				while (parent.getParent() != null && parent.getParent().getNameCount() > 0) {

```

Comment:
can you please add a unit test for that one?

Reply:
I have added it.